flink-libraries/flink-streaming-python/src/main/java/org/apache/flink/streaming/python/api/environment/PythonStreamExecutionEnvironment.java

```diff
@@ -0,0 +1,442 @@
+	 * 		The port of the master (JobManager), where the program should
+	 * 		be executed.
+	 * @param jar_files
+	 * 		The JAR files with code that needs to be shipped to the
+	 * 		cluster. If the program uses user-defined functions,
+	 * 		user-defined input formats, or any libraries, those must be
+	 * 		provided in the JAR files.
+	 * @return A remote environment that executes the program on a cluster.
+	 */
+	public static PythonStreamExecutionEnvironment create_remote_execution_environment(
+		String host, int port, String... jar_files) {
+		return new PythonStreamExecutionEnvironment(host, port, jar_files);
+	}
+
+	/**
+	 * A thin wrapper layer over {@link StreamExecutionEnvironment#createRemoteEnvironment(
+	 * java.lang.String, int, Configuration, java.lang.String...)}
+	 *
+	 * @param host
+	 * 		The host name or address of the master (JobManager), where the
+	 * 		program should be executed.
+	 * @param port
+	 * 		The port of the master (JobManager), where the program should
+	 * 		be executed.
+	 * @param config
+	 * 		The configuration used by the client that connects to the remote cluster.
+	 * @param jar_files
+	 * 		The JAR files with code that needs to be shipped to the
+	 * 		cluster. If the program uses user-defined functions,
+	 * 		user-defined input formats, or any libraries, those must be
+	 * 		provided in the JAR files.
+	 * @return A remote environment that executes the program on a cluster.
+	 *
+	 */
+	public static PythonStreamExecutionEnvironment create_remote_execution_environment(
+		String host, int port, Configuration config, String... jar_files) {
+		return new PythonStreamExecutionEnvironment(host, port, config, jar_files);
+	}
+
+	/**
+	 * A thin wrapper layer over {@link StreamExecutionEnvironment#createRemoteEnvironment(
+	 * java.lang.String, int, int, java.lang.String...)}
+	 *
+	 * @param host
+	 * 		The host name or address of the master (JobManager), where the
+	 * 		program should be executed.
+	 * @param port
+	 * 		The port of the master (JobManager), where the program should
+	 * 		be executed.
+	 * @param parallelism
+	 * 		The parallelism to use during the execution.
+	 * @param jar_files
+	 * 		The JAR files with code that needs to be shipped to the
+	 * 		cluster. If the program uses user-defined functions,
+	 * 		user-defined input formats, or any libraries, those must be
+	 * 		provided in the JAR files.
+	 * @return A remote environment that executes the program on a cluster.
+	 */
+	public static PythonStreamExecutionEnvironment create_remote_execution_environment(
+		String host, int port, int parallelism, String... jar_files) {
+		return new PythonStreamExecutionEnvironment(host, port, parallelism, jar_files);
+	}
+
+	private PythonStreamExecutionEnvironment() {
+		this.env = StreamExecutionEnvironment.getExecutionEnvironment();
+		this.registerJythonSerializers();
+	}
+
+	private PythonStreamExecutionEnvironment(Configuration config) {
+		this.env = new LocalStreamEnvironment(config);
```

Comment:
I would throw out all constructors and add one that takes a `StreamExecutionEnvironmen` parameter. We already have the case distinction in the various `create_*` methods.